### PR TITLE
Added update logic art_galery

### PR DESCRIPTION
**Pull request by Skilskyi**

### CODE REVIEW

File: Streetcode/Streetcode.BLL/DTO/Media/Art/ArtUpdateDto.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using Streetcode.BLL.DTO.Media.Images;
+
+namespace Streetcode.BLL.DTO.Media.Create
```
Comment (MrVenik) [inline Streetcode/Streetcode.BLL/DTO/Media/Art/ArtUpdateDto.cs:3]: File-scoped namespaces please (?)

File: Streetcode/Streetcode.BLL/DTO/Media/Art/ArtUpdateDto.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+{
+    public class ArtUpdateDTO
```
Comment (MrVenik) [inline Streetcode/Streetcode.BLL/DTO/Media/Art/ArtUpdateDto.cs:5]: If this is an request - you can make it record

File: Streetcode/Streetcode.BLL/MediatR/Media/Art/Update/UpdateArtErrorMessages.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Streetcode.BLL.MediatR.Media.Art;
+
+public static class UpdateErrorMessages
```
Comment (MrVenik) [inline Streetcode/Streetcode.BLL/MediatR/Media/Art/Update/UpdateArtErrorMessages.cs:3]: Use Resource files errors messages

File: Streetcode/Streetcode.BLL/MediatR/Media/Art/Update/UpdateArtHandler.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+using Streetcode.BLL.Interfaces.Logging;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+
+namespace Streetcode.BLL.MediatR.Media.Art.Update;
+
+public class UpdateArtHandler : IRequestHandler<UpdateArtCommand, Result<ArtDto>>
+{
+    private readonly IMapper _mapper;
+    private readonly IRepositoryWrapper _repositoryWrapper;
+    private readonly ILoggerService _logger;
```
Comment (MrVenik) [inline Streetcode/Streetcode.BLL/MediatR/Media/Art/Update/UpdateArtHandler.cs:14]: Blank lines between fields and constructors

File: Streetcode/Streetcode.BLL/MediatR/Media/Art/Update/UpdateArtHandler.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    public UpdateArtHandler(
+        IMapper mapper,
+        IRepositoryWrapper repositoryWrapper,
+        ILoggerService logger)
+    {
+        _mapper = mapper;
+        _repositoryWrapper = repositoryWrapper;
+        _logger = logger;
+    }
```
Comment (MrVenik) [inline Streetcode/Streetcode.BLL/MediatR/Media/Art/Update/UpdateArtHandler.cs:23]: same here

File: Streetcode/Streetcode.BLL/MediatR/Media/Art/Update/UpdateArtHandler.cs [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    public async Task<Result<ArtDto>> Handle(UpdateArtCommand request, CancellationToken cancellationToken)
+    {
+        var existingArt = await _repositoryWrapper.ArtRepository
+                .GetFirstOrDefaultAsync(a => a.Id == request.Art.Id);
+
+        if (existingArt is null)
```
Comment (MrVenik) [inline Streetcode/Streetcode.BLL/MediatR/Media/Art/Update/UpdateArtHandler.cs:29]: if existingArt is never used, maybe it is better to left this check for validator (it is for an future takss)

File: Streetcode/Streetcode.WebApi/Controllers/Media/Images/ArtController.cs [hunk 7]
```diff
@@ -25,4 +28,10 @@ public async Task<IActionResult> GetByStreetcodeId([FromRoute] int streetcodeId)
     {
         return HandleResult(await Mediator.Send(new GetArtsByStreetcodeIdQuery(streetcodeId)));
     }
+    [HttpPut("{id:int}")]
+    public async Task<IActionResult> Update([FromRoute] int id, [FromBody] ArtUpdateDTO art)
```
Comment (MrVenik) [inline Streetcode/Streetcode.WebApi/Controllers/Media/Images/ArtController.cs:32]: You can use one request from the form (cause it is an http request from the form)
Why you need this is froum the route ?

**Review by MrVenik — CHANGES_REQUESTED** [review]

**Review by SvyatoS1 — APPROVED** [review]

**Comment by Skilskyi** [comment]
-Removed redundant null check
-Transfered HandlerTest file in correct folder

**Comment by MrVenik** [comment]
Wait untill #132 would be merged, then pull from `story/at_gallery`

**Comment by Skilskyi** [comment]
-Fixed issues with ArtEntity

**Review by MrVenik — APPROVED** [review]
Great job!

**Pull request merged**